wire-compiler/src/main/java/com/squareup/wire/ServiceWriter.java

```diff
@@ -35,5 +35,6 @@ protected ServiceWriter(JavaWriter writer, List<String> options) {
    * @param service the service being generated.
    * @param importedTypes a set of strings containing known imports.
    */
-  public abstract void emitService(Service service, Set<String> importedTypes) throws IOException;
+  public abstract void emitService(ServiceElement service, Set<String> importedTypes)
```

Comment:
Note: this is a public API change, but that's the consequence of exposing a dependencies types. This will have to be a Wire 2.0 if we adhere to semantic versioning.